### ISSUE TITLE
Support configMaps in CSV

### ISSUE THREAD

**Issue by srinivas-kandula** [operator-framework/operator-sdk#2176]
## Feature Request

**Is your feature request related to a problem? Please describe.**
 I am working with RedHat partner connect team to build operator for a application that need to be hosted in Operatorhub.io.  While building CSV using `operator-sdk olm-catalog gen-csv` came across this limitation of operator-sdk.  The operator deploys a application that needs default configMap to be present in the namespace before the application is deployed else the application fails to start.  

**Describe the solution you'd like**
As CSV is having all required metadata for the operator having configMap in the same CSV would make the operator deployment simple.

**Comment by hasbro17** [comment]
@srinivas-kandula I see you've opened up a corresponding issue on OLM as well.
I think that's the correct place to first have the discussion for your use case before we can add this functionality to the SDK.

We need to be clear on what the expected workflow is for requesting OLM or the catalog operator to create an arbitrary required resource like a configmap. Although I suspect that is out of scope for OLM since OLM is only concerned with deploying the operator and the pre-reqs for an application are left up to the user.

In any case I'm going to close this for now so we can follow up on the OLM use case first.

/cc @dmesser
